### ISSUE TITLE
Customizable Fonts

### ISSUE THREAD

**Issue by CryZe** [LiveSplit/livesplit-core#104]
For supporting fonts, we probably want to express them as Font Name + Font Size (and maybe bold, italic, ..., not entirely sure). That should be the most portable representation. I don't think livesplit-core itself needs to be able to provide a list of all the fonts available. I think the UI can do that. For the browser we probably want to just provide a list of common fonts and allow providing a custom font name, as there doesn't seem to be any API (other than a hack through flash) to actually query all of the system's fonts. In the desktop versions we can use some native APIs to query them. So actually, maybe it does make sense to do that in livesplit-core, in a livesplit-font library or something.

**Comment by hallcristobal** [comment]
It might be worth incorporating https://fonts.google.com, or webfontloader https://github.com/typekit/webfontloader for the web versions.

**Comment by wooferzfg** [comment]
Fonts should definitely be expressed with Font Family, Style, and Size.

**Comment by CryZe** [comment]
Considering livesplit-core is handling the rendering now too, we may want to use [font-kit](https://github.com/pcwalton/font-kit) and / or [skribo](https://github.com/linebender/skribo) for loading the fonts, falling back to fonts if they don't have certain glyphs, and properly shaping the glyph runs.

**Comment by CryZe** [comment]
We probably want to abstract them away though, as only for certain operating systems we even want to compile those in. skribo for example has a hard dependency on harfbuzz, which while being a really good library, is a C library and is painful for us to compile, especially to wasm.

**Comment by CryZe** [comment]
Alright so I started working on this. Here's some current random thoughts:

Font loading will be an optional feature (and it's only relevant to when you use the renderer). By default we still ship our default fonts. In the layout there will be font overrides rather than font settings that are always there. font-kit will be used in the renderer if the font loading feature is active. If font override can be found, the font will be used. Otherwise we still fall back to our default font. When importing from *.lsl we detect the default fonts and import them as "not overridden". I noticed that we likely don't want a font size setting at all. However font style (weight, italics, variation axes, ...) is still something that we should have.
